tox.ini

```diff
@@ -0,0 +1,81 @@
+[tox]
+envlist =
+    py{36,37,38}-test{,-alldeps,-devdeps}
+    build_docs
+    linkcheck
+    style
+requires =
+    setuptools >= 30.3.0
+    pip >= 19.3.1
+isolated_build = true
+
+[testenv]
+passenv =
+    HOME
+    WINDIR
+    LC_ALL
+    LC_CTYPE
+    CC
+    CFLAGS
+    CI
+    TRAVIS
+    TEST_READ_HUGE_FILE
+changedir =
+    test: .tmp/{envname}
+    build_docs,linkcheck: docs
+description =
+    test: run tests with pytest
+    build_docs: invoke sphinx-build to build the HTML docs
+    linkcheck: check the links in the HTML docs
+    alldeps: run tests with all dependencies
+    devdeps: run tests with numpy and matplotlib dev versions
+deps =
+    # The oldestdeps tag is intended to be used to install the oldest versions of all
+    # dependencies that have a minimum version
+    oldestdeps: numpy==1.16.*
+    oldestdeps: matplotlib==2.1.*
+    oldestdeps: asdf==2.4.*
+    numpy116: numpy==1.16.*
+    numpy117: numpy==1.17.*
+    numpy118: numpy==1.18.*
+    devdeps: git+https://github.com/numpy/numpy.git#egg=numpy
+    devdeps: git+https://github.com/matplotlib/matplotlib.git#egg=matplotlib
+extras =
+    test: test,cov
+    build_docs,linkcheck: docs
+    alldeps: all
+commands =
+    pip freeze
+    test: pytest --pyargs astropy {toxinidir}/docs --cov astropy --cov-config={toxinidir}/astropy/tests/coveragerc {posargs}
+    build_docs: sphinx-build -W -b html . _build/html {posargs}
+    linkcheck: sphinx-build -W -b linkcheck . _build/html {posargs}
+
+[testenv:style]
+# PEP8 errors/warnings:
+# E101 - mix of tabs and spaces
+# W191 - use of tabs
+# W291 - trailing whitespace
+# W292 - no newline at end of file
+# W293 - trailing whitespace
+# W391 - blank line at end of file
+# E111 - 4 spaces per indentation level
+# E112 - 4 spaces per indentation level
+# E113 - 4 spaces per indentation level
+# E301 - expected 1 blank line, found 0
+# E302 - expected 2 blank lines, found 0
+# E303 - too many blank lines (3)
+# E304 - blank lines found after function decorator
+# E305 - expected 2 blank lines after class or function definition
+# E306 - expected 1 blank line before a nested definition
+# E502 - the backslash is redundant between brackets
+# E722 - do not use bare except
+# E901 - SyntaxError or IndentationError
+# E902 - IOError
+# E999: SyntaxError -- failed to compile a file into an Abstract Syntax Tree
+# # F821: undefined name  # Note: Removed for now because of heavy use of units.si
+# F822: undefined name in __all__
+# F823: local variable name referenced before assignment
+skip_install = true
+description = check code style, e.g. with flake8
+deps = flake8
+commands = flake8 astropy --count --select=E101,W191,W291,W292,W293,W391,E111,E112,E113,E30,E502,E722,E901,E902,E999,F822,F823
```

Comment:
I think the selection should be taken from/kept inside `setup.cfg`, so that `flake8` picks up on it (and emacs too!)

Reply:
This would shadow other useful errors/warnings in your editor. There is an issue open about this, and a consensus on reducing the number of errors/warnings first before moving to excluding instead of selecting the codes.

Reply:
Yes indeed as @saimn pointed out, this is deliberate and is why we used to define this list in the Travis configuration. The real fix here is to simply fix all flake8 warnings in the astropy code base so that things just work with the bare flake8 command.

Reply:
I've added a comment to explain this in tox.ini